pkg/service/service.go

```diff
@@ -1369,17 +1370,23 @@ func (s *Service) restoreServicesLocked() error {
 		if option.Config.DatapathMode == datapathOpt.DatapathModeLBOnly &&
 			newSVC.useMaglev() && recreated {
 
-			backends := make(map[string]lb.BackendID, len(newSVC.backends))
-			for _, b := range newSVC.backends {
-				backends[b.String()] = b.ID
-			}
-			if err := s.lbmap.UpsertMaglevLookupTable(uint16(newSVC.frontend.ID), backends,
-				ipv6 || svc.NatPolicy == lb.SVCNatPolicyNat46); err != nil {
-				scopedLog.WithError(err).Warning("Unable to upsert into the Maglev BPF map.")
-				continue
+			_, activeBackends, _ := segregateBackends(newSVC.backends)
+			if len(activeBackends) > 0 {
+				if err := s.lbmap.UpsertMaglevLookupTable(uint16(newSVC.frontend.ID), activeBackends,
+					ipv6 || svc.NatPolicy == lb.SVCNatPolicyNat46); err != nil {
+					scopedLog.WithError(err).Warning("Unable to upsert into the Maglev BPF map.")
+					continue
+				}
+				if svc.NatPolicy == lb.SVCNatPolicyNat46 {
```

Comment:
Why is this if-branch needed?

Reply:
my bad during rebasing.. let me fix that